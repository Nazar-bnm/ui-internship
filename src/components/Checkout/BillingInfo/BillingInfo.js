import React from 'react';
import cx from 'classnames';

import { Dropdown, Button } from '@/shared';
import dropdownOptions from '../../../../config';

import './BillingInfo.scss';

const CN = 'billing-info';

// This function should be replaced when the proper functional for dropdowns is added.
const onDropdownChange = () => true;

const BillingInfo = () => (
  <div className={cx(CN)}>
    <form className={`${CN}__form`}>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="first-name">first name*</label>
        <input type="text" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="last-name">last name*</label>
        <input type="text" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="company">Company</label>
        <input type="text" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="e-mail">e-mail address*</label>
        <input type="email" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="address1">address 1</label>
        <input type="text" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="address2">address 2</label>
        <input type="text" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <p>country</p>
        <Dropdown
          menuOptions={dropdownOptions.billingInfo.country}
          value="Choose country..."
          onDropdownChange={onDropdownChange}
          className={`${CN}__dropdown`}
        />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="city">city</label>
        <input type="text" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <p>state</p>
        <Dropdown
          menuOptions={dropdownOptions.billingInfo.state}
          value="Choose state..."
          onDropdownChange={onDropdownChange}
          className={`${CN}__dropdown`}
        />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="zipcode">zip / postal code*</label>
        <input type="number" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="phone">phone</label>
        <input type="number" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="fax">fax</label>
        <input type="number" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="password">password*</label>
        <input type="password" name={CN} className={`${CN}__input`} />
      </div>
      <div className={`${CN}__input-wrapper`}>
        <label htmlFor="password">confirm password*</label>
        <input type="password" name={CN} className={`${CN}__input`} />
      </div>
    </form>
    <form className={`${CN}__radiobtns-form`}>
      <div className={`${CN}__radiobtn`}>
        <input className={`${CN}__radiobtn-input`} type="radio" name="ship-to-this-address" value="ship-to-this-address" />
        <label htmlFor="ship-to-this-address">Ship to this address</label>
      </div>
      <div className={`${CN}__radiobtn`}>
        <input className={`${CN}__radiobtn-input`} type="radio" name="ship-to-this-address" value="ship-to-different-address" />
        <label htmlFor="guest">Ship to a different address</label>
      </div>
    </form>
    <div className={`${CN}__checkbox-wrapper`}>
      <input className={`${CN}__radiobtn-input`} type="checkbox" />
          I want to subscribe to the newsletter
    </div>
    <Button customClass="login-register__button">continue</Button>
  </div>
);

export default BillingInfo;
