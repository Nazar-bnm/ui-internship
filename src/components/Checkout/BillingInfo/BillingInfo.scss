@import '../../../scss/global_styles';

.billing-info {
  @include flex($direction: column);
  padding: 20px;
  font-family: font(primary-light);
  text-transform: uppercase;

  &__form {
    @include flex($direction: column, $justify: space-between);
    width: 100%;
  }

  &__radiobtns-form {
    @include flex;
    height: 50px;
    margin: 10px 0;
  }

  &__radiobtn {
    @include flex($align: center);

    &-input {
      margin-right: 5px;
    }
  }

  &__checkbox-wrapper {
    @include flex($align: center);
    margin: 10px 0;
  }

  &__input {
    margin: 5px 0;
    padding: 5px;
    width: 100%;
    border: 1px solid $silver-chalice;
  }

  .dropdown {
    display: block;
    position: relative;
    padding: 5px;
    margin: 5px 0;
    color: $black;
    border: 1px solid $silver-chalice;
    background-color: $white;

    .caret {
      position: absolute;
      right: 0;
      top: 50%;
    }

    &__items {
      top: 105%;
      width: 100%;

      &__element {
        text-transform: uppercase;
      }
    }
  }

  .dropdown[aria-expanded=true] .dropdown__items {
    top: 105%;
    border: 2px solid $silver-chalice;
    background-color: $alabaster;
  }
}

@media only screen and (min-width: 480px) {
  .billing-info {
    &__form {
      flex-direction: row;
    }

    &__radiobtns-form {
      flex-direction: column;
      justify-content: space-evenly;
    }

    &__checkbox-wrapper {
      margin: 0 10px 20px 0;
    }
    
    &__input-wrapper {
      width: 45%;
    }
  }
}