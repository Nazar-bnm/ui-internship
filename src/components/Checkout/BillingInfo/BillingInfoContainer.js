import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import BillingInfo from './BillingInfo';
import showMessage from '../../../actions/actionsNotifications';

const mapDispatchToProps = (dispatch) => ({
  showMessage: bindActionCreators(showMessage, dispatch)
});

export default connect(null, mapDispatchToProps)(BillingInfo);
