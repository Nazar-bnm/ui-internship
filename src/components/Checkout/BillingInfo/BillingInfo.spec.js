import React from 'react';
import { shallow } from 'enzyme';

import BillingInfo from './BillingInfo';

describe('<BillingInfo />', () => {
  let wrapper = shallow(<BillingInfo />);
  
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change region state when select a region', () => {
    const event = 'hello';
    wrapper = shallow(<BillingInfo />);
    wrapper.find('RegionDropdown').simulate('change', event);
    expect(wrapper.state().region).toBe(event);
  })

  test('should change country state when select a country', () => {
    const event = 'hello';
    wrapper = shallow(<BillingInfo />);
    wrapper.find('CountryDropdown').simulate('change', event);
    expect(wrapper.state().country).toBe(event);
  })
});
