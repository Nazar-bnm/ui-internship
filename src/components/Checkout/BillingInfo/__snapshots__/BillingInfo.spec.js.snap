// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BillingInfo /> should match snapshot 1`] = `
<div
  className="billing-info"
>
  <form
    className="billing-info__form"
  >
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="first-name"
      >
        first name*
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="text"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="last-name"
      >
        last name*
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="text"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="company"
      >
        company
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="text"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="e-mail"
      >
        e-mail address*
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="email"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="address1"
      >
        address 1
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="text"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="address2"
      >
        address 2
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="text"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <p>
        country
      </p>
      <Dropdown
        className="billing-info__dropdown"
        menuOptions={
          Array [
            Object {
              "label": "Ukraine",
              "value": "Ukraine",
            },
            Object {
              "label": "Poland",
              "value": "Poland",
            },
            Object {
              "label": "USA",
              "value": "USA",
            },
            Object {
              "label": "China",
              "value": "China",
            },
            Object {
              "label": "France",
              "value": "France",
            },
          ]
        }
        onDropdownChange={[Function]}
        placeholder=""
        value="Choose country..."
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="city"
      >
        city
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="text"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <p>
        state
      </p>
      <Dropdown
        className="billing-info__dropdown"
        menuOptions={
          Array [
            Object {
              "label": "--not applicable--",
              "value": "--not applicable--",
            },
            Object {
              "label": "Alabama",
              "value": "Alabama",
            },
            Object {
              "label": "Alaska",
              "value": "Alaska",
            },
            Object {
              "label": "Arizona",
              "value": "Arizona",
            },
            Object {
              "label": "Arkansas",
              "value": "Arkansas",
            },
            Object {
              "label": "California",
              "value": "California",
            },
          ]
        }
        onDropdownChange={[Function]}
        placeholder=""
        value="Choose state..."
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="zipcode"
      >
        zip / postal code*
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="number"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="phone"
      >
        phone
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="number"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="fax"
      >
        fax
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="number"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="password"
      >
        password*
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="password"
      />
    </div>
    <div
      className="billing-info__input-wrapper"
    >
      <label
        htmlFor="password"
      >
        confirm password*
      </label>
      <input
        className="billing-info__input"
        name="billing-info"
        type="password"
      />
    </div>
  </form>
  <form
    className="billing-info__radiobtns-form"
  >
    <div
      className="billing-info__radiobtn"
    >
      <input
        className="billing-info__radiobtn-input"
        name="ship-to-this-address"
        type="radio"
        value="ship-to-this-address"
      />
      <label
        htmlFor="ship-to-this-address"
      >
        Ship to this address
      </label>
    </div>
    <div
      className="billing-info__radiobtn"
    >
      <input
        className="billing-info__radiobtn-input"
        name="ship-to-this-address"
        type="radio"
        value="ship-to-different-address"
      />
      <label
        htmlFor="guest"
      >
        Ship to a different address
      </label>
    </div>
  </form>
  <div
    className="billing-info__checkbox-wrapper"
  >
    <input
      className="billing-info__radiobtn-input"
      type="checkbox"
    />
    I want to subscribe to the newsletter
  </div>
  <Button
    customClass="black-button"
    icon={null}
    onClick={[Function]}
  >
    continue
  </Button>
</div>
`;
